sycl/include/sycl/detail/ur.hpp

```diff
@@ -0,0 +1,315 @@
+/// native asserts if supported extensions' names are requested
+#define UR_DEVICE_INFO_EXTENSION_DEVICELIB_ASSERT "cl_intel_devicelib_assert"
+
+/// Device binary image property set names recognized by the SYCL runtime.
+/// Name must be consistent with
+/// PropertySetRegistry::SYCL_SPECIALIZATION_CONSTANTS defined in
+/// PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SPEC_CONST_MAP "SYCL/specialization constants"
+/// PropertySetRegistry::SYCL_SPEC_CONSTANTS_DEFAULT_VALUES defined in
+/// PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SPEC_CONST_DEFAULT_VALUES_MAP                   \
+  "SYCL/specialization constants default values"
+/// PropertySetRegistry::SYCL_DEVICELIB_REQ_MASK defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_DEVICELIB_REQ_MASK "SYCL/devicelib req mask"
+/// PropertySetRegistry::SYCL_KERNEL_PARAM_OPT_INFO defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_KERNEL_PARAM_OPT_INFO "SYCL/kernel param opt"
+/// PropertySetRegistry::SYCL_KERNEL_PROGRAM_METADATA defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_PROGRAM_METADATA "SYCL/program metadata"
+/// PropertySetRegistry::SYCL_MISC_PROP defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_MISC_PROP "SYCL/misc properties"
+/// PropertySetRegistry::SYCL_ASSERT_USED defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_ASSERT_USED "SYCL/assert used"
+/// PropertySetRegistry::SYCL_EXPORTED_SYMBOLS defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_EXPORTED_SYMBOLS "SYCL/exported symbols"
+/// PropertySetRegistry::SYCL_DEVICE_GLOBALS defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_DEVICE_GLOBALS "SYCL/device globals"
+/// PropertySetRegistry::SYCL_DEVICE_REQUIREMENTS defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_DEVICE_REQUIREMENTS                        \
+  "SYCL/device requirements"
+/// PropertySetRegistry::SYCL_HOST_PIPES defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_HOST_PIPES "SYCL/host pipes"
+/// PropertySetRegistry::SYCL_VIRTUAL_FUNCTIONS defined in PropertySetIO.h
+#define __SYCL_UR_PROPERTY_SET_SYCL_VIRTUAL_FUNCTIONS "SYCL/virtual functions"
+
+/// Program metadata tags recognized by the UR backends. For kernels the tag
+/// must appear after the kernel name.
+#define __SYCL_UR_PROGRAM_METADATA_TAG_REQD_WORK_GROUP_SIZE                    \
+  "@reqd_work_group_size"
+#define __SYCL_UR_PROGRAM_METADATA_GLOBAL_ID_MAPPING "@global_id_mapping"
+
+#define __SYCL_UR_PROGRAM_METADATA_TAG_NEED_FINALIZATION "Requires finalization"
+
+/// This struct is a record of the device binary information. If the Kind field
+/// denotes a portable binary type (SPIR-V or LLVM IR), the DeviceTargetSpec
+/// field can still be specific and denote e.g. FPGA target. It must match the
+/// __tgt_device_image structure generated by the clang-offload-wrapper tool
+/// when their Version field match.
+struct ur_device_binary_struct {
```

Comment:
This is not a UR device binary struct, this is our compiler device binary struct. I think more and more that names should be neutralized (i.e. they should be neither `pi_` nor `ur_` and this should be outlined into a separate header about our interface between compiler and runtime as a separate PR.

Reply:
This is also addressed in #14700